feat: Daily Habits Tab w/ User Data

calebpanza opened the pull request:
## ✏️ Description
This PR introduces the final components and user data connected to the Daily Tab.

## 🔬 To Test
1. Set the app to connect to the Production dB
2. Load up the "Daily" tab
3. Follow the prompts to walk through a complete Daily Experience.

## 📸 Screenshots

## ⚠️ To-do Before Merge


## 📝 Checklist
- [ ] I have performed a self-review of my code
- [ ] I have tested my code works in the project (nothing breaks)
- [ ] I have commented my code in hard-to-understand areas
- [ ] New and existing tests and linting pass locally with my changes
- [ ] I have made corresponding changes to the documentation (new envs?)

jordanwade commented:
@calebpanza I'm see are error with `status` on the Daily tab.
<img width="425" alt="image" src="https://user-images.githubusercontent.com/2528817/202021281-387c92e1-f038-4812-acfe-68f3fa56f125.png">

calebpanza commented:
@jordanwade can you pull changes? I fixed that a few commits back

calebpanza commented:
https://github.com/ApollosProject/apollos-apps/pull/2766/commits/378fcffb8a3e910e25fa7848bf1b207b90ebf091

jordanwade commented:
<img width="393" alt="image" src="https://user-images.githubusercontent.com/2528817/202023106-94caa64c-eafc-4240-9b2f-108e35abe0d7.png">
After filling out what I'm grateful for I see this pop up blank.

calebpanza commented:
@jordanwade for that experience, I have a separate F&I list. I connected the tab to the existing Experiences that need to be re-tested.

This PR is only for the tab and new components.

jordanwade commented:
@calebpanza sorry, I must be misunderstanding what I should be testing/reviewing. I'm not seeing the "mon" through "sun" component and I cannot seem to get the state of anything to update. 

Is it possible I'm pointed at the wrong `APP_DATA_URL`? `https://apollos-cluster-production.herokuapp.com/`

<img width="367" alt="image" src="https://user-images.githubusercontent.com/2528817/202026014-77cef765-0e57-453d-a956-cde79cceb71d.png">

calebpanza commented:
OH.. that's 10000% my bad. @jordanwade I didn't update the testing instructions properly before making this ready for review. It's up to date now

jordanwade commented:
@calebpanza I still can't get the state to change but I'll just ping you in slack, instead of blowing up your PR more 😄

calebpanza commented:
@jordanwade OK sounds good! We can talk tomorrow then. In the meantime, I posted a video on loom which provides a more in depth showing of what you can expect to be testing and stuff which hopefully helps too! For all future PR's, I'll make sure there are videos attached with screenshots.

camrun91 commented:
The pulse card causes the app to crash for me you said something about it possibly having something to do with the profile image.

camrun91 commented:
@calebpanza I am having a hard time testing this as well I can't actually post anything to get the state to change.

calebpanza commented:
@camrun91 @jordanwade I found that the issue was some kind of compatibility problem with the `react-native-image-filter-kit` and iOS 15. In order to just get this moving, I've removed that package and replaced all the gradients rendered with `react-native-linear-gradient`.

Hopefully this resolves any of the crashing issues! And we should be good to just point to Production now for the `APP_DATA_URL`!

camrun91 reviewed (approved):
This one looks good to me with the changes you just made.

calebpanza commented:
Oh thank GOD! Hahah
